File: 42588.java
class Solution {
    public int[] solution(int[] heights) {
        int len = heights.length;
        int[] answer = new int[len];
        
        for(int i = len-1; i > -1 ; i--){
            // System.out.println(heights[i]);
            int item = heights[i];
            for(int j = i ; j > -1 ; j--){
                if(heights[j] > item){
                    answer[i] = j+1;
                    break;
                }
            }
     
        }
        return answer;
    }
}

File: 42588.swift
import Foundation

func solution(_ heights:[Int]) -> [Int] {
    var answer = Array(repeating: 0, count: heights.count)

    for i in (0..<heights.count).reversed(){
        for j in (0..<i).reversed(){
            if heights[i] < heights[j] {
                // print("i : \(heights[i]) , j : \(heights[j])")
                answer[i] = j+1
                break
            }
            
        }
    }
    return answer
}
